test/k8sT/DatapathConfiguration.go

```diff
@@ -295,6 +319,21 @@ var _ = Describe("K8sDatapathConfig", func() {
 			Expect(testPodConnectivityAcrossNodes(kubectl)).Should(BeTrue(), "Connectivity test between nodes failed")
 		})
 
+		SkipItIf(helpers.DoesNotRunOnNetNextOr419Kernel, "Check BPF masquerading", func() {
+			defaultIface, err := kubectl.GetDefaultIface()
+			Expect(err).Should(BeNil(), "Failed to retrieve default iface")
+
+			deployCilium(map[string]string{
+				"global.bpfMasquerade":        "true",
+				"global.nodePort.device":      fmt.Sprintf(`'{%s,%s}'`, privateIface, defaultIface),
+				"global.tunnel":               "disabled",
+				"global.autoDirectNodeRoutes": "true",
```

Comment:
Could we also add a test when XDP acceleration is enabled as well to make sure both don't break when enabled at the same time?

Reply:
Btw, should we get the small fix in #10874 resolved first to make sure we have proper CI coverage for the new tests on 4.19?

Reply:
Re testing with XDP: how flaky is the XDP test? Maybe worth waiting until #10942 is resolved to avoid adding new flakes?

Re 4.19: if #10874 gets merged soon (hopefully the last commit has fixed the issue), then yeah, we can wait. Anyway, I'd prefer to get this PR (#10878) merged sooner than later, as rebasing && making sure that CI is green is taking too much time and effort.